x/liquidstakeibc/module_ibc.go

```diff
@@ -24,38 +24,95 @@ func NewIBCModule(keeper keeper.Keeper) IBCModule {
+		portID,
+		channelID,
+		channelCap,
+		counterparty,
+		version,
+	)
 }
 
-func (ibcModule IBCModule) OnChanOpenTry(ctx sdk.Context, order channeltypes.Order, connectionHops []string, portID, channelID string, channelCap *capabilitytypes.Capability, counterparty channeltypes.Counterparty, counterpartyVersion string) (version string, err error) {
-	return "", nil
+func (m IBCModule) OnChanOpenAck(
+	ctx sdk.Context,
+	portID string,
+	channelID string,
+	counterpartyChannelID string,
+	counterpartyVersion string,
+) error {
+	return m.keeper.OnChanOpenAck(
+		ctx,
+		portID,
+		channelID,
+		counterpartyChannelID,
+		counterpartyVersion,
+	)
 }
 
-func (ibcModule IBCModule) OnChanOpenAck(ctx sdk.Context, portID, channelID string, counterpartyChannelID string, counterpartyVersion string) error {
-	return ibcModule.keeper.OnChanOpenAck()
+func (m IBCModule) OnAcknowledgementPacket(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	acknowledgement []byte,
+	relayer sdk.AccAddress,
+) error {
+	return m.keeper.OnAcknowledgementPacket(
+		ctx,
+		packet,
+		acknowledgement,
+		relayer,
+	)
 }
 
-func (ibcModule IBCModule) OnChanOpenConfirm(ctx sdk.Context, portID, channelID string) error {
-	return nil
+func (m IBCModule) OnTimeoutPacket(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	relayer sdk.AccAddress,
+) error {
+	return m.keeper.OnTimeoutPacket(
+		ctx,
+		packet,
+		relayer,
+	)
 }
 
-func (ibcModule IBCModule) OnChanCloseInit(ctx sdk.Context, portID, channelID string) error {
-	return nil
+func (m IBCModule) OnChanOpenTry(
+	ctx sdk.Context,
+	order channeltypes.Order,
+	connectionHops []string,
+	portID string,
+	channelID string,
+	channelCap *capabilitytypes.Capability,
+	counterparty channeltypes.Counterparty,
+	counterpartyVersion string,
+) (version string, err error) {
+	panic("UNIMPLEMENTED")
 }
 
-func (ibcModule IBCModule) OnChanCloseConfirm(ctx sdk.Context, portID, channelID string) error {
-	return nil
+func (m IBCModule) OnChanOpenConfirm(ctx sdk.Context, portID, channelID string) error {
+	panic("UNIMPLEMENTED")
 }
 
-func (ibcModule IBCModule) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, relayer sdk.AccAddress) ibcexported.Acknowledgement {
-	return nil
+func (m IBCModule) OnChanCloseInit(ctx sdk.Context, portID, channelID string) error {
+	panic("UNIMPLEMENTED")
 }
 
-func (ibcModule IBCModule) OnAcknowledgementPacket(ctx sdk.Context, packet channeltypes.Packet, acknowledgement []byte, relayer sdk.AccAddress) error {
-	return ibcModule.keeper.OnAcknowledgementPacket()
+func (m IBCModule) OnChanCloseConfirm(ctx sdk.Context, portID, channelID string) error {
+	panic("UNIMPLEMENTED")
 }
 
-func (ibcModule IBCModule) OnTimeoutPacket(ctx sdk.Context, packet channeltypes.Packet, relayer sdk.AccAddress) error {
-	return ibcModule.keeper.OnTimeoutPacket()
+func (m IBCModule) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, relayer sdk.AccAddress) ibcexported.Acknowledgement {
+	panic("UNIMPLEMENTED")
```

Comment:
why panics here? code probably will never reach here since this is controller wrapper.. 
usually just follow ibc-go controller handling for this

Reply:
any preference over returning nil ? I saw panicking as a visual way of mentioning this was not implemented